lib/config/configure.js

```diff
@@ -91,7 +91,7 @@ module.exports = _.curry((rascalConfig, next) => {
     } = new URL(connectionString);
     const options = Array.from(searchParams).reduce((attributes, entry) => ({ ...attributes, [entry[0]]: entry[1] }), {});
     return {
-      protocol, hostname, port, user, password, vhost, options,
+      protocol, hostname: decodeURIComponent(hostname), port, user: decodeURIComponent(user), password: decodeURIComponent(password), vhost: decodeURIComponent(vhost), options,
```

Comment:
There is reason I wanted to have a config for the consumer to indicate what was already encoded in https://github.com/onebeyond/rascal/pull/228. What if there is a sequence in the `user` like `%23` that was not a product of encoding, but just part of an actual password? This code here will destructively decode it.

Reply:
Ha! I see you're testing that [below](https://github.com/onebeyond/rascal/pull/229/files#diff-59cc4a3af1181865c858de26aa6d223f2c332351d061d6e6025d24fcd190bfb5R83). Have to think about it a bit more to see if there is a case of a destructive decoding...

Reply:
> Ha! I see you're testing that [below](https://github.com/onebeyond/rascal/pull/229/files#diff-59cc4a3af1181865c858de26aa6d223f2c332351d061d6e6025d24fcd190bfb5R83). Have to think about it a bit more to see if there is a case of a destructive decoding...

No, I'm confusing myself. I think my original idea in the above PR is correct.

There is no way for rascal to know whether the consumer intended `#` as the character, or the set of `%23` to be in the password (or the other parts of the url).

What your test assumes is that a consumer that passes a url like

```
http://friend:%23Mellon@Khazad-dum.west.gate
```

doesn't actually want rascal to encode it to 

```
http://friend:%2523Mellon@Khazad-dum.west.gate
``` 

Does that make sense?

Reply:
I would also argue this is a breaking change